xdr/common.go

```diff
@@ -40,11 +43,121 @@ func HashFromSlice(slice []byte) (Hash, error) {
 	return fromSlice32(slice)
 }
 
-// IdFromPublicKey : from generic crypto.PublicKey interface{} we try to extract an ID
-func IdFromPublicKey(pk crypto.PublicKey) (ID, error) {
+// IDFromPublicKey : from generic crypto.PublicKey interface{} we try to extract an ID
+func IDFromPublicKey(pk crypto.PublicKey) (ID, error) {
 	bbytes, ok := pk.([]byte)
 	if ok == false {
 		return ID{}, errors.New("public key not a slice of bytes")
 	}
-	return IdFromSlice(bbytes)
+	return IDFromSlice(bbytes)
+}
+
+// MarshalJSON : when marshaling to JSON we want ID to be represented as a base64-encoded string, not as an array of uint8
+func (id *ID) MarshalJSON() ([]byte, error) {
```

Comment:
All of the MarshalJSON functions need to use non pointer receivers `(id ID)` in order to work with json.Marshal and when included in other objects such as ChannelConfig.

With this if you do a json.Marshal(id) or json.Marshal(channelConfig) it will not call the MarshalJSON function on the id field.